### PR TITLE
Run `nox` test sessions in parallel

### PR DESCRIPTION
**Pull request by grahamalama**
This PR specifies a `matrix` for our text workflow where each option of the matrix is a test session as defined in Nox.

Running tests in parallel cuts the workflow time in half:

Before:
image

After
image

We could potentially get this number down further with caching, but this is a nice improvement for now.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,12 +11,37 @@ on:
...
+    outputs:
+      sessions: ${{ steps.set-sessions.outputs.sessions }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      - run: pip install nox
+
+      - name: List Nox sessions
```
Comment (hackebrot) [inline .github/workflows/test.yml:25]: Can you describe the purpose of this step? I don't see `nox_sessions.json` referenced here in the workflow.

**Review by hackebrot — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (grahamalama) [reply .github/workflows/test.yml]: Oops! The description is, didn't clean up after myself after listing sessions and saving that output in the same step.

78ef01ab1d733c6ae246528e6dff25060f016afc

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -11,12 +11,32 @@ on:
...
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'      
+      - run: pip install nox
+      - name: List and set Nox sessions matrix
+        id: set-sessions
+        run: |
+          sessions=$(nox --list --tag "test" --json | jq -r '. | map("\"" + .session + "\"") | join(",")')
+          echo "sessions=[$sessions]" >> $GITHUB_OUTPUT
```
Comment (hackebrot) [inline .github/workflows/test.yml:28]: Can we simplify this to the following? Grabbing the `session` field of each object and returning valid JSON directly?

```suggestion
          sessions=$(nox --list --tag "test" --json | jq '[ .[] | .session ]')
          echo "sessions=$sessions" >> $GITHUB_OUTPUT
```

**Review by hackebrot — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (grahamalama) [reply .github/workflows/test.yml]: Oh good call. I found I had to add the `--compact-output` flag as well, or else I got [an error](https://github.com/grahamalama/fourkeys/actions/runs/11114716192/job/30881735319)

Comment (hackebrot) [reply .github/workflows/test.yml]: 👍🏻

**Review by hackebrot — APPROVED** [review]
LGTM! 🚀

**Pull request merged**